clinvar_this/io/gks_json.py

```diff
@@ -0,0 +1,390 @@
+        )
+
+    @staticmethod
+    def _get_variant_coords(
+        variant: Variation,
+    ) -> SubmissionChromosomeCoordinates | None:
+        """Get the chromosome coordinates for a variant
+
+        Looks in the variant's extensions to find an extension with the name
+        'CIViC representative coordinate'
+
+        :param variant: Variant record from CIViC
+        :return: Chromosome coordinates for a variant, if found
+        """
+        try:
+            rep_coords_ext = next(
+                ext
+                for ext in variant.extensions
+                if ext.name == "CIViC representative coordinate"
+            ).value
+        except StopIteration:
+            return
+
+        return SubmissionChromosomeCoordinates(
+            assembly=Assembly(rep_coords_ext["reference_build"]),
+            chromosome=Chromosome(rep_coords_ext["chromosome"]),
+            start=rep_coords_ext["start"],
+            stop=rep_coords_ext["stop"],
+        )
+
+    @staticmethod
+    def _get_citations(
+        mp_id: str,
+        evidence_lines: typing.List[EvidenceLine],
+    ) -> typing.List[SubmissionCitation]:
+        """Get list of PubMed citations and supporting CIViC evidence for an assertion
+
+        :param mp_id: CIViC molecular profile ID for assertion
+        :param evidence_lines: A list of evidence-based arguments that may support or
+            refute the validity of a proposition for a CIViC assertion
+        :return: List of PubMed citations and supporting CIViC evidence URLs for an
+            assertion
+        """
+        citations = [SubmissionCitation(url=f"https://identifiers.org/{mp_id}")]
+        for evidence_line in evidence_lines:
+            for evidence_item in getattr(evidence_line, "hasEvidenceItems", []):
+                citations.append(
+                    SubmissionCitation(
+                        url=f"https://civicdb.org/links/evidence/{evidence_item.id.split('civic.eid:')[-1]}"
+                    )
+                )
+
+                docs = evidence_item.reportedIn or []
+                for doc in docs:
+                    if getattr(doc, "pmid", None):
+                        citations.append(
+                            SubmissionCitation(db=CitationDb.PUBMED, id=str(doc.pmid))
+                        )
+
+        return citations
+
+    @staticmethod
+    def _get_drugs(therapeutic: TherapyGroup | MappableConcept) -> str:
+        """Get the name for drug(s)
+
+        :param therapeutic: CIViC therapeutic record
+        :return: The formatted name for a CIViC therapeutic record. Multiple therapies
+            for combination and substitution will be separated by semicolons.
+        """
+        if isinstance(therapeutic, MappableConcept):
+            return therapeutic.name
+
+        return ";".join([t.name for t in therapeutic.therapies])
+
+    @staticmethod
+    def _get_last_accepted_revision(record_id: int) -> str | None:
```

Comment:
Check to see what CIViC does

Reply:
<img width="1057" alt="Screenshot 2025-02-21 at 09 58 27" src="https://github.com/user-attachments/assets/b5f735f9-796a-43fc-ae84-4e6443707c40" />

![Screenshot 2025-02-21 at 09 58 53](https://github.com/user-attachments/assets/b23b19f4-291e-4233-8a25-ff57e916f03b)

Looks like submittedEvent was used

Reply:
Arpad said: 

> I would generally use the accepted date for this. but if there were major revisions I would then look at the activity feed for that

I'm following up on this in Slack. Will report back later.